From 172e1f5ec0e37c8aab91a2ae35bd73ea594432cb Mon Sep 17 00:00:00 2001
Message-Id: <172e1f5ec0e37c8aab91a2ae35bd73ea594432cb.1571920849.git.pmatilai@redhat.com>
In-Reply-To: <6b6c4d881dc6fc99f949dac4aaf9a513542f9956.1571920849.git.pmatilai@redhat.com>
References: <6b6c4d881dc6fc99f949dac4aaf9a513542f9956.1571920849.git.pmatilai@redhat.com>
From: Panu Matilainen <pmatilai@redhat.com>
Date: Thu, 4 Oct 2018 13:36:09 +0300
Subject: [PATCH 2/5] Use Python 3 -compatible exception syntax in tests

Makes a few tests pass that failed before, and others now fail
a little bit later...

(cherry picked from commit 511eef19298765e3639bccbe98bc3a50023f45b2)
---
 tests/rpmpython.at | 12 ++++++------
 1 file changed, 6 insertions(+), 6 deletions(-)

diff --git a/tests/rpmpython.at b/tests/rpmpython.at
index 3a7c251f1..1daaf1216 100644
--- a/tests/rpmpython.at
+++ b/tests/rpmpython.at
@@ -96,7 +96,7 @@ for a in ['name', 'bugurl', '__class__', '__foo__', ]:
     try:
         x = getattr(h, a)
         myprint(x)
-    except AttributeError, exc:
+    except AttributeError as exc:
         myprint(exc)
 ],
 [testpkg-5:1.0-1.noarch
@@ -119,7 +119,7 @@ h2['dirindexes'] = [ 0, 0, 1 ]
 for h in [h1, h2]:
     try:
         myprint(','.join(h['filenames']))
-    except rpm.error, exc:
+    except rpm.error as exc:
         myprint(exc)
 ],
 [invalid header data
@@ -164,7 +164,7 @@ rpm.setLogFile(sink)
 try:
     h = ts.hdrFromFdno('${RPMDATA}/RPMS/hello-2.0-1.x86_64-signed.rpm')
     myprint(h['arch'])
-except rpm.error, e:
+except rpm.error as e:
     myprint(e)
 ],
 [public key not available
@@ -183,7 +183,7 @@ ts.setKeyring(keyring)
 try:
     h = ts.hdrFromFdno('${RPMDATA}/RPMS/hello-2.0-1.x86_64-signed.rpm')
     myprint(h['arch'])
-except rpm.error, e:
+except rpm.error as e:
     myprint(e)
 ],
 [x86_64]
@@ -207,7 +207,7 @@ h = rpm.hdr()
 h['name'] = "foo"
 try:
     ts.addInstall(h, 'foo', 'u')
-except rpm.error, err:
+except rpm.error as err:
     myprint(err)
 for e in ts:
     myprint(e.NEVRA())
@@ -228,7 +228,7 @@ h['dirnames'] = ['/opt' '/flopt']
 h['dirindexes'] = [ 1, 2, 3 ]
 try:
     ts.addInstall(h, 'foo', 'u')
-except rpm.error, err:
+except rpm.error as err:
     myprint(err)
 for e in ts:
     myprint(e.NEVRA())
-- 
2.21.0

